code/datums/components/clothing_dirt.dm

```diff
@@ -0,0 +1,90 @@
+/// This component applies tint to clothing when its exposed to pepperspray, used in /obj/item/clothing/mask/gas.
+
+/datum/component/clothing_dirt
+	/// The ITEM_SLOT_* slot the item is equipped on, if it is.
+	var/equipped_slot
+
+	/// Mob wearing the clothing
+	var/mob/living/carbon/wearer
+
+	/// Amount of dirt stacks on the clothing
+	var/dirtiness = 0
+
+	/// Clothing we're applying tint on
+	var/obj/item/clothing/clothing
```

Comment:
Don't have this extra var that points to parent. Just cast parent to clothing when necessary
```dm
/datum/component/clothing_dirt/UnregisterFromParent()
	var/obj/item/clothing/clothing = parent
	clothing.tint -= dirtiness
	clothing = null